kfs-ld/src/main/java/edu/arizona/kfs/module/ld/batch/service/impl/SalaryTransferPeriodValidationServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ * 
+ * Copyright 2005-2014 The Kuali Foundation
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.arizona.kfs.module.ld.batch.service.impl;
+
+import edu.arizona.kfs.gl.service.GlobalTransactionEditService;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.integration.ec.EffortCertificationReport;
+import org.kuali.kfs.module.ld.LaborKeyConstants;
+import org.kuali.kfs.module.ld.businessobject.ExpenseTransferAccountingLine;
+import org.kuali.kfs.module.ld.document.SalaryExpenseTransferDocument;
+import org.kuali.kfs.module.ld.document.service.SalaryTransferPeriodValidationService;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.Message;
+import org.kuali.kfs.sys.businessobject.AccountingLineBase;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.krad.util.GlobalVariables;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @see SalaryTransferPeriodValidationService
+ */
+@Transactional
+public class SalaryTransferPeriodValidationServiceImpl extends org.kuali.kfs.module.ld.document.service.impl.SalaryTransferPeriodValidationServiceImpl {
```

Comment:
The way the directory structure displayed in the IDE got me, I saw "blah.service.impl" in the collapsed structure, and missed that the prefix was "batch", not "document". Corrected.